Added PK Restrictions

Arkhist opened the pull request:
If a player kills somebody 6 levels under itself, the target won't loose XP or drop his inventory.

Commit:
Made gitignore and removed classpath, to allow commits

Commit:
Added PK restriction

Titousensei commented on SERVER/src/data_handlers/battle_handler/HitHandler.java:
```diff
@@ -411,11 +411,14 @@ public static void targetDeath(Creature TARGET, Creature ATTACKER, boolean arena
       PlayerCharacter playerTarget = (PlayerCharacter) TARGET;
 
-      boolean pkAttack = false;
+      int pkAttack = 0;
```
I'm not a big fan of int as a state indicator. Please create an enum instead.

Arkhist replied:
Enum added

Titousensei commented on SERVER/src/data_handlers/battle_handler/HitHandler.java:
```diff
@@ -411,11 +411,14 @@ public static void targetDeath(Creature TARGET, Creature ATTACKER, boolean arena
         if (ATTACKER.getCreatureType() == CreatureType.Player) {
-          pkAttack = true;
+          pkAttack = 1;
+          if(ATTACKER.getLevel() > TARGET.getLevel()+5)
```
Number 5 is used in 2 places. Please create a final static int with ALL_CAPS name because it's a constant.

Arkhist replied:
I needed to make two final static int because it's used in Server and Client. I'm not fond of placing it in "Common".

Titousensei commented on CLIENT/src/creature/PlayerCharacter.java:
```diff
@@ -192,6 +192,9 @@ public void draw(Graphics g, int centerX, int centerY) {
         g.drawString(Name, nameX, cornerY - 25);
+        
+        if(BlueSaga.playerCharacter.getLevel() > this.getLevel()+5)
+        	g.drawString("Weak", centerX - Font.size10.getWidth("Weak"), cornerY-12);
```
The centering calculation does not work here because you forgot to /2.
I would prefer an icon next to the name, but we can change that later.

Arkhist replied:
I'm very bad in drawing icons, will keep the text.

Titousensei commented on SERVER/src/data_handlers/battle_handler/BattleHandler.java:
```diff
@@ -654,7 +654,8 @@ public static void playerDeath(Client client, boolean pkAttack) {
 
       // CHECK IF PLAYER CAN WEAR EQUIP, OTHERWISE DROP IT ON GROUND
-      ItemHandler.loseLootUponDeath(client);
+      if(pkAttack != 2)
```
If you do an enum, you can replace those 2 ifs with a switch and make the code easier to read.

Arkhist replied:
Not fond of using a switch in this context.

Commit:
Added PlayerDeathCause enum,
added static int PK_WEAK_LEVEL clientside & serverside,
fixed rendering of "Weak" status

Titousensei reviewed (approved):
Values in Settings classes should not be final, so they can be overridden at runtime with .cfg files. I will make the change.
